File: problems/Check If a String Contains All Binary Codes of Size K/hasAllCodes.cs
public class Solution {
    public bool HasAllCodes(string s, int k) {
        var store = new HashSet<string>();

		if (0 > s.Length - k) {
            return false;
        }

		for (var i = 0; s.Length - k >= i; ++i) {
            var str = s.Substring(i, k);

            if (!store.Contains(str)) {
                store.Add(str);
            }
        }

		return store.Count == Math.Pow(2, k);
    }
}

File: problems/Check If a String Contains All Binary Codes of Size K/hasAllCodes.java
class Solution {
    public boolean hasAllCodes(String s, int k) {
        Set<String> store = new HashSet<>();

        for (int idx = 0; s.length() - k >= idx; ++idx) {
            store.add(s.substring(idx, idx + k));
        }

        return store.size() == Math.pow(2, k);
    }
}
